GG/GG/Font.h

```diff
@@ -147,6 +151,13 @@ class GG_API Font
             pair.second. */
         Substring(const std::string& str_, const IterPair& pair);
 
+        /** Attach this Substring to \p str_.
+
+            This changes the iterators from pointing into the previous
```

Comment:
The iterator is invalid because it does not point to the string the its own `SubString` is pointing to.  If it still points to a string that has not been de-allocated, then dereferencing it won't cause a crash, but it is still invalid.

So with multiple Substrings all pointing to the same string, rebinding one Substring does not invalidate any iterators from the other Substrings.  It does invalidate the iterators from the rebound Substring, but happenstance prevents dereferencing those invalid iterators from causing a segmentation fault.